Description: Fix the parsing and writing of wep-hex configurations
 Symptom: Can't connect to WEP based networks with hexadecimal keys
 due to unexpanded template variables like $_KEY in the actual
 wpa_supplicant configurations.
 .
 Solution: Move the appending of the line outside the loop over all
 matches in one line.
Author: Axel Beckert <abe@debian.org>
Bug-Debian: https://bugs.debian.org/812427
Bug: https://bugs.launchpad.net/wicd/+bug/1419464
Forwarded: https://bugs.launchpad.net/wicd/+bug/1419464/comments/16

Index: wicd/wicd/misc.py
===================================================================
--- wicd.orig/wicd/misc.py	2016-01-24 01:49:06.625462455 +0100
+++ wicd/wicd/misc.py	2016-01-24 01:51:38.042711547 +0100
@@ -321,11 +321,11 @@
                                 rep_val = '0'
                         if rep_val:
                             line = line.replace("$_%s" % cur_val, str(rep_val))
-                            config_file = ''.join([config_file, line])
                         else:
                             print "Ignoring template line: '%s'" % line
                     else:
                         print "Weird parsing error occurred"
+                config_file = ''.join([config_file, line])
             else:  # Just a regular entry.
                 config_file = ''.join([config_file, line])
 
